pkg/reconciler/pipeline/controller.go

```diff
@@ -845,37 +845,49 @@ func (r *pipelineReconciler) pausePipeline(ctx context.Context, pl *dfv1.Pipelin
 		return updated, err
 	}
 
-	daemonClient, err := daemonclient.NewGRPCDaemonServiceClient(pl.GetDaemonServiceURL())
+	var daemonError error
+	var drainCompleted = false
+
+	// Check for the daemon to obtain the buffer draining information, in case we see an error trying to
+	// retrieve this we do not exit prematurely to allow honoring the pause timeout for a consistent error
+	// - In case the timeout has not occurred we would trigger a requeue
+	// - If the timeout has occurred even after getting the drained error, we will try to pause the pipeline
+	daemonClient, daemonError := daemonclient.NewGRPCDaemonServiceClient(pl.GetDaemonServiceURL())
+	r.logger.Info(daemonClient.ListPipelineBuffers(ctx, pl.Name))
 	if err != nil {
-		return true, err
+		daemonError = err
```

Comment:
so, if I'm reading this correctly, this is the `err` from the call above `r.scaleDownSourceVertices()`, right? Is that a daemon error?

Reply:
but actually, it looks like we return above if `err != nil` so I believe this code is unreachable ?